packages/xrpl/tools/generateModels.js

```diff
@@ -0,0 +1,265 @@
+export { ${tx} } from './${tx}'`,
+  )
+  fs.writeFileSync(filename, indexFile)
+}
+
+function generateParamLine(sfields, param, isRequired) {
+  const paramName = param.slice(2)
+  const paramType = sfields[paramName][0]
+  const paramTypeOutput = typeMap[paramType]
+  return `  ${paramName}${isRequired ? '' : '?'}: ${paramTypeOutput}\n`
+}
+
+async function main(folder) {
+  const [txsToAdd, txFormats, sfields, transactionMatch] =
+    processRippledSource(folder)
+  txsToAdd.forEach(async (tx) => {
+    const txFormat = txFormats[tx]
+    const paramLines = txFormat
+      .filter((param) => param[0] !== '')
+      .sort((a, b) => a[0].localeCompare(b[0]))
+      .map((param) =>
+        generateParamLine(sfields, param[0], param[1] === 'soeREQUIRED'),
+      )
+    paramLines.sort((a, b) => !a.includes('REQUIRED'))
+    const params = paramLines.join('\n')
+    let model = `/**
+ * @category Transaction Models
+ */
+export interface ${tx} extends BaseTransaction {
+  TransactionType: '${tx}'
+
+${params}
+}`
+
+    const commonImports = []
+    const validationImports = ['BaseTransaction', 'validateBaseTransaction']
+    for (const item of allCommonImports) {
+      if (params.includes(item)) {
+        commonImports.push(item)
+        validationImports.push('is' + item)
+      }
+    }
+    for (const item of additionalValidationImports) {
+      if (params.includes(item)) {
+        validationImports.push(
+          'is' + item.substring(0, 1).toUpperCase() + item.substring(1),
+        )
+      }
+    }
+    if (params.includes('?')) {
+      validationImports.push('validateOptionalField')
+    }
+    if (/[A-Za-z0-9]+:/.test(params)) {
+      validationImports.push('validateRequiredField')
+    }
+    validationImports.sort()
+    const commonImportLine =
+      commonImports.length > 0
+        ? `import { ${commonImports.join(', ')} } from '../common'`
+        : ''
+    const validationImportLine = `import { ${validationImports.join(
+      ', ',
+    )} } from './common'`
+    let imported_models = `${commonImportLine}
+
+${validationImportLine}`
+    imported_models = imported_models.replace('\n\n\n\n', '\n\n')
+    imported_models = imported_models.replace('\n\n\n', '\n\n')
+    model = model.replace('\n\n\n\n', '\n\n')
+    fs.writeFileSync(
+      path.join(
+        path.dirname(__filename),
+        `../src/models/transactions/${tx}.ts`,
+      ),
+      imported_models + '\n\n' + model,
+    )
+
+    const validate = await createValidate(tx)
+    fs.appendFileSync(
+      path.join(
+        path.dirname(__filename),
+        `../src/models/transactions/${tx}.ts`,
+      ),
+      '\n\n' + validate,
+    )
+
+    const validateTests = createValidateTests(tx)
+    fs.writeFileSync(
+      path.join(path.dirname(__filename), `../test/models/${tx}.test.ts`),
+      validateTests,
+    )
+
+    updateTransactionFile(transactionMatch, tx)
+
+    updateIndexFile(tx)
+
+    console.log('Added ' + tx)
```

Comment:
Might be worth adding a note on what the next steps that people have to do by hand would be :)
```suggestion
    const pathname = path/to/new/model
    ...
    console.log(`Added  ${tx} to ${pathname}`)
    console.log(`Next you should: \n  1. Add docstrings to the model fields.\n  2. Add an integration test for the model like ${pathToExample}`)  
```